rust/cubesql/cubesql/src/compile/rewrite/rules/split.rs

```diff
@@ -1544,6 +1680,77 @@ impl RewriteRules for SplitRules {
                 agg_fun_expr("?fun", vec![literal_expr("?literal")], "?distinct"),
                 self.transform_outer_aggr_fun_missing_count("?cube", "?fun"),
             ),
+            // Aggregate function with cast argument
+            rewrite(
+                "split-push-down-aggr-fun-cast-arg-float64-inner-replacer",
+                inner_aggregate_split_replacer(
+                    agg_fun_expr(
+                        "?fun",
+                        vec![cast_expr_explicit(
+                            column_expr("?column"),
+                            ArrowDataType::Float64,
+                        )],
+                        "?distinct",
+                    ),
+                    "?alias_to_cube",
+                ),
+                inner_aggregate_split_replacer(
+                    agg_fun_expr("?fun", vec![column_expr("?column")], "?distinct"),
+                    "?alias_to_cube",
+                ),
+            ),
+            transforming_chain_rewrite(
+                "split-push-down-aggr-fun-cast-arg-float64-outer-replacer",
+                outer_projection_split_replacer("?expr", "?alias_to_cube"),
+                vec![(
+                    "?expr",
+                    agg_fun_expr(
```

Comment:
I think outer_projection_split_replacer can not contains agg_fun_expr

Reply:
Lines L421-437 and L1614-1623 didn't have the problem with agg_fun_expr inside outer_projection. We're moving the cast here and it only affects Float64, so this should be bijective.